lib/invitations/acceptInvitation.ts

```diff
@@ -0,0 +1,144 @@
+import { prisma } from "@lib/integration/prismaConnector";
+import { FormProperties, UserAbility } from "@lib/types";
+import {
+  InvitationIsExpiredError,
+  InvitationNotFoundError,
+  TemplateNotFoundError,
+  UserNotFoundError,
+} from "./exceptions";
+import { checkPrivileges } from "@lib/privileges";
+import { ownerAddedNotification } from "@lib/invitations/emailTemplates/ownerAddedNotification";
+import { sendEmail } from "@lib/integration/notifyConnector";
+import { logEvent } from "@lib/auditLogs";
+
+/**
+ * Accept an invitation.
+ * User has created their account or logged into their existing account.
+ *
+ * @param ability (logged in user)
+ * @param invitationId
+ * @returns
+ */
+export const acceptInvitation = async (ability: UserAbility, invitationId: string) => {
+  // Retrieve the invitation
```

Comment:
Wondering if we should grab the ability from 

```
const { ability } = await authCheckAndThrow();
```

vs passing it in ....

Reply:
The `authCheckAndThrow()` and `authCheckAndRedirect()` functions should only be used in Server Components because it leverages React cache.  Usage outside of a Server Component causes the main auth function to run on each call and recreates an ability and session.  I should update the comments in that code file and ensure doc strings are available that explain the caveats.